Core/Core/ActAsUser/ActAsUserViewController.swift

```diff
@@ -0,0 +1,129 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+import UIKit
+
+public protocol ActAsUserPresenterProtocol: class {
+    func didSubmit(domain: String, userID: String, completion: @escaping (Error?) -> Void)
+}
+
+public class ActAsUserViewController: UIViewController {
+    var env: AppEnvironment?
+    var presenter: ActAsUserPresenterProtocol?
+
+    @IBOutlet var redPanda: UIImageView!
+    @IBOutlet var redPandaTopConstraint: NSLayoutConstraint!
+    @IBOutlet var redPandaTrailingConstraint: NSLayoutConstraint!
+    @IBOutlet var actAsUserDescription: UILabel!
+    @IBOutlet var domainTextField: UITextField!
+    @IBOutlet var userIDTextField: UITextField!
+    @IBOutlet var actAsUserButton: UIButton!
+
+    public static func create(env: AppEnvironment = .shared, presenter: ActAsUserPresenterProtocol) -> ActAsUserViewController {
+        let controller = self.loadFromStoryboard()
+        controller.env = env
+        controller.presenter = presenter
+        return controller
+    }
+
+    override public func viewDidLoad() {
+        addCancelButton()
+        title = NSLocalizedString("Act as User", bundle: .core, comment: "")
+        // swiftlint:disable:next line_length
+        actAsUserDescription.text = NSLocalizedString("\"Act as\" is essentially logging in as this user without a password. You will be able to take any action as if you were this user, and from other users' points of view, as if this user performed them. However, audit logs record that you were the one who performed the actions on behalf of this user.", bundle: .core, comment: "")
+        domainTextField.placeholder = NSLocalizedString("Domain", bundle: .core, comment: "")
+        userIDTextField.placeholder = NSLocalizedString("User ID", bundle: .core, comment: "")
+        actAsUserButton.titleLabel?.text = NSLocalizedString("Act as User", bundle: .core, comment: "")
+        domainTextField.text = env?.currentSession?.baseURL.absoluteString
+
+        domainTextField.addTarget(self, action: #selector(updateActAsUserButtonDisabledStatus), for: .editingChanged)
+        userIDTextField.addTarget(self, action: #selector(updateActAsUserButtonDisabledStatus), for: .editingChanged)
+        actAsUserButton.isEnabled = false
+
+        domainTextField.addTarget(self, action: #selector(enterPressed), for: .editingDidEndOnExit)
+        userIDTextField.addTarget(self, action: #selector(enterPressed), for: .editingDidEndOnExit)
+
+        redPanda.transform = redPanda.transform.rotated(by: .pi / -6).translatedBy(x: 0, y: 0)
+        actAsUserButton.roundCorners(corners: [.topLeft, .topRight, .bottomLeft, .bottomRight], radius: 5)
```

Comment:
if you are rounding all of the corners `layer.cornerRadius = 5` is probably simpler. I'm not sure about performance difference. That can also be added as a custom property in the storyboard.

Reply:
I was trying to do `cornerRadius` but it was saying it was not a property on a `UIButton`. Didn't realize I had to do it on the `layer`. I will make that change.